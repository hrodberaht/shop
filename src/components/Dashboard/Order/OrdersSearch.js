import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class OrdersSearch extends Component {
  static propTypes = {
    handleSelect: PropTypes.func.isRequired,
  };

  render() {
    const { handleSelect } = this.props;
    return (
      <div>
        {'Orders: '}
        <select onChange={handleSelect}>
          <option value="all">All</option>
          <option value="realized">Realized</option>
          <option value="in-progress">In progress</option>
        </select>
      </div>
    );
  }
}
