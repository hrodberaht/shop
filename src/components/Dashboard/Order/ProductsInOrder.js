import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ProductsInOrder extends Component {
  render() {
    const {
      order: { productsOrder },
    } = this.props;
    return (
      <td colSpan="5">
        <table>
          <tbody>
            <tr>
              <th>Name:</th>
              <th>Pcs:</th>
              <th>Price</th>
              <th>Total price:</th>
            </tr>
            {productsOrder.map((product) => {
              const {
                id, name, price, pcsOrder, totalPrice,
              } = product;
              return (
                <tr key={id}>
                  <td>{name}</td>
                  <td>{pcsOrder}</td>
                  <td>{price}</td>
                  <td>{totalPrice}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </td>
    );
  }
}

ProductsInOrder.propTypes = {
  order: PropTypes.objectOf(PropTypes.arrayOf).isRequired,
};
