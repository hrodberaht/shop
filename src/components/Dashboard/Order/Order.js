import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown, faAngleUp, faShippingFast } from '@fortawesome/free-solid-svg-icons';
import ProductsInOrder from './ProductsInOrder';
import ConnectedAuthorization from '../../Auth/Authorization';
import priceFix from '../../../shared/priceFix';

export default class Order extends Component {
  state = {
    details: false,
  };

  handleClickDetails = () => {
    this.setState(state => ({ details: !state.details }));
  };

  render() {
    const { details } = this.state;
    const {
      handleClick,
      token,
      order,
      order: {
        id, person, totalPrice, status, date,
      },
    } = this.props;
    return (
      <React.Fragment>
        <tr>
          <td>{id}</td>
          <td>{person}</td>
          <td>{date}</td>
          <td>{priceFix(totalPrice)}</td>
          <td>{status}</td>
          <td>
            <button type="button" className="btn-orders-detail" onClick={this.handleClickDetails}>
              <FontAwesomeIcon icon={details ? faAngleUp : faAngleDown} />
            </button>
          </td>
          <ConnectedAuthorization
            render
            withRoleAdmin={(
              <td className="orders_realized">
                <button
                  className="btn btn-submit"
                  type="button"
                  disabled={order.status === 'realized'}
                  onClick={() => handleClick(order.id, token)}
                >
                  <FontAwesomeIcon icon={faShippingFast} />
                </button>
              </td>
)}
          />
        </tr>
        {details && (
          <tr key={order.date}>
            <ProductsInOrder order={order} />
          </tr>
        )}
      </React.Fragment>
    );
  }
}

Order.propTypes = {
  order: PropTypes.shape({
    id: PropTypes.string,
    userId: PropTypes.string,
    data: PropTypes.string,
    totalPrice: PropTypes.number,
    status: PropTypes.string,
  }).isRequired,
  handleClick: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
};
