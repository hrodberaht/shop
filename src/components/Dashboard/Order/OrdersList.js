import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getOrders } from '../../../store/orders/selectors';
import Order from './Order';
import ConnectedAuthorization from '../../Auth/Authorization';
import { changeOrderStatus } from '../../../store/orders/actionCreator';

export class OrdersList extends Component {
  render() {
    const { orders, handleClick } = this.props;
    return (
      <div>
        <table>
          <tbody>
            <tr>
              <th>Id:</th>
              <th>Person:</th>
              <th>Date:</th>
              <th>Total price:</th>
              <th>Status:</th>
              <ConnectedAuthorization render withRoleAdmin={<th>Change status:</th>} />
            </tr>
            {orders.map(order => (
              <tr key={order.id}>
                <Order order={order} />
                <ConnectedAuthorization
                  render
                  withRoleAdmin={(
                    <td>
                      <button type="button" onClick={() => handleClick(order.id)}>
                        Realise
                      </button>
                    </td>
)}
                />
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  orders: getOrders(state),
});

export default connect(
  mapStateToProps,
  { handleClick: changeOrderStatus },
)(OrdersList);

OrdersList.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleClick: PropTypes.func.isRequired,
};
