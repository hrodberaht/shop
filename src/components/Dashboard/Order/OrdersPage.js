import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getOrders, getLoadedStatus } from '../../../store/orders/selectors';
import ConnectedAuthorization from '../../Auth/Authorization';

import { fetchOrders, fetchChangeOrderStatus } from '../../../store/orders/actionCreators';
import { getAuthToken, getAuthUserId } from '../../../store/authenticate/selectors';
import OrdersSearch from './OrdersSearch';
import OrdersList from './OrdersList';

export class OrdersPage extends Component {
  state = {
    status: 'all',
  };

  componentDidMount() {
    const { getOrdersFromServer, token, userId } = this.props;
    getOrdersFromServer(userId, token);
  }

  handleSelect = (e) => {
    this.setState({ status: e.target.value });
  };

  render() {
    const { orders, loaded } = this.props;
    if (!loaded) return <p>Loading...</p>;
    return (
      <React.Fragment>
        <div>
          <OrdersSearch handleSelect={this.handleSelect} />
        </div>
        <table>
          <tbody>
            <tr className="order-table-title">
              <th>Id:</th>
              <th>Person:</th>
              <th>Date:</th>
              <th>Total price:</th>
              <th>Status:</th>
              <th>Actions:</th>
              <ConnectedAuthorization render withRoleAdmin={<th>Admin:</th>} />
            </tr>
            <OrdersList filters={this.state.status} orders={orders} />
          </tbody>
        </table>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  token: getAuthToken(state),
  orders: getOrders(state),
  loaded: getLoadedStatus(state),
  userId: getAuthUserId(state),
});

export default connect(
  mapStateToProps,
  {
    handleClick: fetchChangeOrderStatus,
    getOrdersFromServer: fetchOrders,
  },
)(OrdersPage);

OrdersPage.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  loaded: PropTypes.bool.isRequired,
  getOrdersFromServer: PropTypes.func.isRequired,
  token: PropTypes.string,
  userId: PropTypes.string,
};

OrdersPage.defaultProps = {
  token: null,
  userId: null,
};
