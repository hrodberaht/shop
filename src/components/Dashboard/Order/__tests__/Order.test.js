import Order from '../Order';

const defaultProps = {
  order: {
    id: '1234',
    userId: '2345',
    data: '12.13.12',
    totalPrice: 400,
    status: 'in-progress',
  },
};
const setup = buildSetup(Order, defaultProps);

describe('<Order />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
