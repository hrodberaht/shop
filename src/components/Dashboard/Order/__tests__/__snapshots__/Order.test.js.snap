// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> should render without crash 1`] = `
<Fragment>
  <tr>
    <td>
      1234
    </td>
    <td />
    <td />
    <td>
      400
    </td>
    <td>
      in-progress
    </td>
    <td>
      <button
        onClick={[Function]}
        type="button"
      >
        \\/
      </button>
    </td>
    <td>
      <Connect(Authorization)
        render={true}
        withRoleAdmin={
          <button
            className="btn btn-submit"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Realise
          </button>
        }
      />
    </td>
  </tr>
</Fragment>
`;
